frontend/packages/policy-editor/src/utils/index.ts

```diff
@@ -231,8 +230,7 @@ export const mergeSubjectsFromPolicyWithSubjectOptions = (
   rules.forEach((rule) => {
     rule.subject.forEach((subjectString) => {
       const subjectId = convertSubjectStringToSubjectId(subjectString);
-
-      if (!existingSubjectIds.includes(subjectId)) {
+      if (!existingSubjectIds.includes(subjectId?.toLowerCase())) {
```

Comment:
The return type for `convertSubjectStringToSubjectId` is string, do we need the null-check here? If we need it, should the return type of `convertSubjectStringToSubjectId` be `string | undefined`? 🤔 

```suggestion
      if (!existingSubjectIds.includes(subjectId.toLowerCase())) {
```